web-prover-core/src/parser/config.rs

```diff
@@ -0,0 +1,224 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::parser::{errors::ExtractorError, extractor, extractor::ExtractionResult, Extractor};
+
+/// The format of the data to extract from
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
+#[serde(rename_all = "lowercase")]
+pub enum DataFormat {
+  /// JSON format
+  #[default]
+  Json,
+}
+
+/// The root configuration for data extractors
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
+pub struct ExtractorConfig {
+  /// The format of the data to extract from
+  pub format:     DataFormat,
```

Comment:
thinking out loud: can we deduce this from the path? like `a.0.b.c` for json and `/html/body/div[2]/article/div[4]/p[21]/b/strong` for xpath?

Reply:
I'd prefer to be explicit where possible (but not to the point of redundancy). Nothing is set in stone, perhaps the work on HTML format may influece that #547